### PR TITLE
Grpc variations

### PR DESCRIPTION
**Pull request by sriumcp**
## Main changes
* Values used in templates are now clearly typed

## Other changes
* CI / Makefile improvements
* Test improvements
* Documentation improvements
  * values.yaml documented as per Helm's instructions

### CODE REVIEW

File: mkdocs/docs/tutorials/load-test-http/usage.md
Comment (kalantar) [inline mkdocs/docs/tutorials/load-test-http/usage.md]: Instead of a second `string` header, perhaps `Content type`

File: mkdocs/docs/tutorials/load-test-http/usage.md
Comment (kalantar) [inline mkdocs/docs/tutorials/load-test-http/usage.md]: Perhaps it would be helpful if the names the user should use in the command were explicit here. The example shows several of them and perhaps it is possible to guess the rest. I would like to see them in the list.

File: README.md
Comment (kalantar) [inline README.md:61]: The links in the example have changed; should be updated.

File: hub/load-test-grpc/README.md
Comment (kalantar) [inline hub/load-test-grpc/README.md:4]: These links have changed. Should be updated.

File: hub/load-test-http/README.md
Comment (kalantar) [inline hub/load-test-http/README.md:9]: The links have changed; should be updated.

File: hub/load-test-http/values.yaml
Comment (kalantar) [inline hub/load-test-http/values.yaml]: Per discussion in https://github.com/iter8-tools/iter8/issues/1164 `numQueries` cannot be defaulted to 100; this will override any duration the user sets.

File: mkdocs/docs/getting-started/concepts.md
Comment (kalantar) [inline mkdocs/docs/getting-started/concepts.md]: "safely rollout winning version" should be "safely rollout **a** winning version"

File: mkdocs/docs/getting-started/concepts.md
Comment (kalantar) [inline mkdocs/docs/getting-started/concepts.md:20]: The link to the example should be updated.

File: mkdocs/docs/getting-started/install.md
Comment (kalantar) [inline mkdocs/docs/getting-started/install.md:29]: What is our dependency on go 1.17?

File: mkdocs/docs/getting-started/your-first-experiment.md
Comment (kalantar) [inline mkdocs/docs/getting-started/your-first-experiment.md:56]: I wonder if we've made this too simple by removing the SLOs. I believe SLOs are a core feature (see "Features at a glance"). I believe we should have at least one SLO in this example. It does not make it conceptually more difficult, IMO.

File: mkdocs/docs/getting-started/your-first-experiment.md
Comment (kalantar) [inline mkdocs/docs/getting-started/your-first-experiment.md:159]: I think something like this is valuable on the first experiment page. Maybe "Next Steps" or "Try Next".  It gives direction to the user about where to go. It does not have to be as complete as the current list.

File: mkdocs/docs/tutorials/load-test-grpc/appserverlessml.md
Comment (kalantar) [inline mkdocs/docs/tutorials/load-test-grpc/appserverlessml.md:38]: I find this too terse.  This may  be a first entry for a knative user. I think it would be helpful to put some basic text about what the experiment is doing. A link could be included to the usage examples page as well.

File: mkdocs/docs/tutorials/load-test-http/appserverlessml.md
Comment (kalantar) [inline mkdocs/docs/tutorials/load-test-http/appserverlessml.md:31]: same comment about terseness as in http tests

**Review by kalantar — CHANGES_REQUESTED** [review]

File: mkdocs/docs/tutorials/load-test-http/usage.md
Comment (sriumcp) [reply mkdocs/docs/tutorials/load-test-http/usage.md]: Done.

File: mkdocs/docs/tutorials/load-test-http/usage.md
Comment (sriumcp) [reply mkdocs/docs/tutorials/load-test-http/usage.md]: Done.

File: README.md
Comment (sriumcp) [reply README.md:61]: Will update this README once these changes are merged and we cut a release.

File: hub/load-test-grpc/README.md
Comment (sriumcp) [reply hub/load-test-grpc/README.md:4]: Will fix once we cut a release. Note that mkdocs has the appropriate redirects for them already, so they will not be broken.

File: hub/load-test-http/README.md
Comment (sriumcp) [reply hub/load-test-http/README.md:9]: mkdocs has appropriate redirects for them. So, they are not broken links... will change the actual links after we cut this release.

File: mkdocs/docs/getting-started/install.md
Comment (sriumcp) [reply mkdocs/docs/getting-started/install.md:29]: sigs.k8s.io/yaml v1.3.0 -- seems to complain for go 1.16. Note that for an end-user using brew, there is probably no go dependency; MacOS binary will be installed. 

For an enduser installing Iter8 using binaries, there is no go dependency.

For developers building from source, or for end-users using `go install`, go1.17 is required.

File: mkdocs/docs/getting-started/your-first-experiment.md
Comment (sriumcp) [reply mkdocs/docs/getting-started/your-first-experiment.md:56]: I agree SLOs are a core feature.

However, the first experiment does provide value without SLOs already. In particular, it is benchmarking an application and reporting metrics information.

***

With v0.9, let us explore your second experiment along the following lines... 

```yaml
- your second experiment
  - with SLOs # with some basic description of what SLOs are
  - inside GitHub Actions
  - inside Tekton
  - with notifications
  - with rollouts
  - inside Kubernetes
  - with AutoX
```

All of the above can be part of the getting started page as well ... idea is they are mostly agnostic to the experiment type (load-test-http, load-test-grpc, mirroring, etc.)

File: mkdocs/docs/tutorials/load-test-grpc/appserverlessml.md
Comment (sriumcp) [reply mkdocs/docs/tutorials/load-test-grpc/appserverlessml.md:38]: Could we please revisit this in v0.9? I would like your first Knative experiment to run inside K8s actually (so that it can support both external and cluster local ksvcs).

File: mkdocs/docs/tutorials/load-test-http/appserverlessml.md
Comment (sriumcp) [reply mkdocs/docs/tutorials/load-test-http/appserverlessml.md:31]: Could we please revisit this in v0.9? I would like your first Knative experiment to run inside K8s actually (so that it can support both external and cluster local ksvcs).

File: mkdocs/docs/getting-started/your-first-experiment.md
Comment (sriumcp) [reply mkdocs/docs/getting-started/your-first-experiment.md:159]: Could we hold off on this for a bit? `Your second experiment` might address many of these concerns.

File: mkdocs/docs/tutorials/load-test-grpc/appserverlessml.md
Comment (kalantar) [reply mkdocs/docs/tutorials/load-test-grpc/appserverlessml.md:38]: is this not the example that the current knative doc points to?  I would imagine we want to keep it more clear for any users who to try it.

File: mkdocs/docs/getting-started/your-first-experiment.md
Comment (kalantar) [reply mkdocs/docs/getting-started/your-first-experiment.md:56]: this is great. i don't understand why we are cutting everything out in advance of that. between now and then users have to piece it together on their own. i see greater value in the simple example that includes slos

File: hub/load-test-grpc/README.md
Comment (kalantar) [reply hub/load-test-grpc/README.md:4]: i do see the redirects. for my own understanding, can you explain why waiting for release is a better option. i don't understand why wait ... and be forced to release 2x.

File: mkdocs/docs/getting-started/concepts.md
Comment (sriumcp) [reply mkdocs/docs/getting-started/concepts.md:20]: The link that it should be updated with ... will come into existence after we do this merge. Hence, the need for staging it.

File: hub/load-test-grpc/README.md
Comment (sriumcp) [reply hub/load-test-grpc/README.md:4]: Could we please handle the business of the links in a subsequent PR?

***

The repo structure, with code and docs together, and the practice of updating them both in the same PR, doesn't seem to be doing us favors.

File: mkdocs/docs/getting-started/your-first-experiment.md
Comment (sriumcp) [reply mkdocs/docs/getting-started/your-first-experiment.md:56]: Your first experiment is not meant to teach everything about Iter8 in one go. It is meant to get the user to run the simplest possible experiment to get started with Iter8. The section is titled "getting started", and this is the simplest possible start.
***
The idea is to keep your first experiment, as simple as possible ...

File: mkdocs/docs/tutorials/load-test-grpc/appserverlessml.md
Comment (sriumcp) [reply mkdocs/docs/tutorials/load-test-grpc/appserverlessml.md:38]: I agree. Will push a change to address this.

File: hub/load-test-grpc/README.md
Comment (sriumcp) [reply hub/load-test-grpc/README.md:4]: Updating the README links before the release would mean there is a period of time (between merge and release), when the projects README page has broken links.
***
I agree with thee point on `forced release 2x`. This is a bigger problem due to repo structure... which we will handle better in v0.9

File: hub/load-test-http/README.md
Comment (sriumcp) [reply hub/load-test-http/README.md:9]: Please see README comment earlier on updating links prematurely before they come into existence.

File: hub/load-test-http/values.yaml
Comment (sriumcp) [reply hub/load-test-http/values.yaml]: Done.

File: mkdocs/docs/getting-started/concepts.md
Comment (sriumcp) [reply mkdocs/docs/getting-started/concepts.md]: Done.

File: mkdocs/docs/getting-started/your-first-experiment.md
Comment (sriumcp) [reply mkdocs/docs/getting-started/your-first-experiment.md:159]: Done.

File: mkdocs/docs/tutorials/load-test-grpc/appserverlessml.md
Comment (sriumcp) [reply mkdocs/docs/tutorials/load-test-grpc/appserverlessml.md:38]: Done.

File: mkdocs/docs/tutorials/load-test-http/appserverlessml.md
Comment (sriumcp) [reply mkdocs/docs/tutorials/load-test-http/appserverlessml.md:31]: Done.

File: mkdocs/docs/getting-started/your-first-experiment.md
Comment (kalantar) [reply mkdocs/docs/getting-started/your-first-experiment.md:56]: ok. i see a couple of next steps that attracted my attention. i think this is good

**Review by kalantar — APPROVED** [review]
lgtm

**Pull request merged**
